### PR TITLE
chore: documentation of data sources with ssl

### PR DESCRIPTION
**Pull request by starbirdtech383**
## Change Summary
documentation for ssl options configuration for adding data source

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [x] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/noco-docs/docs/100.data-sources/020.connect-to-data-source.md [hunk 1]
```diff
@@ -23,6 +23,14 @@ To connect to an external data source, follow the steps below:
    | Database      | Name of the database to connect to                                                   |
    | Schema name   | Name of the schema to connect to                                                     |
 
+4a. Optionally, if the connection required is TLS/MTLS for MITM protection, follow these additional steps below:
+
+   - Click on `SSL & Advanced Parameters`, \
```
Comment (wingkwong) [inline packages/noco-docs/docs/100.data-sources/020.connect-to-data-source.md:28]: 1. there is a unncessary `\`
2. `4a.` seems not required

image

File: packages/noco-docs/docs/100.data-sources/020.connect-to-data-source.md [hunk 2]
```diff
@@ -23,6 +23,14 @@ To connect to an external data source, follow the steps below:
...
+   - Select `SSL Mode` and upload the client certificate, client key, and Root CA files by clicking on the file.    
+   \
+   Note: The entire connection parameters can be viewed in json form by clicking on the "Edit connection JSON" and edited as required by database server type.\
+   Example: In PostgreSQL with SSL mode set to "Required-Identity," if the server certificate's common name (cname) differs from the actual DNS/IP used for connection, the connection may fail. To resolve this, add a "servername" property with the cname value under the SSL section, along with certificate information. Additional details are available in the Knex configuration options.[knex configuration options](https://knexjs.org/guide/#configuration-options)
```
Comment (wingkwong) [inline packages/noco-docs/docs/100.data-sources/020.connect-to-data-source.md:32]: 1. better have a line break between note and example. they look too packed.
2. `options.[knex configuration options](https://knexjs.org/guide/#configuration-options)` . Maybe just `[Knex configuration options](https://knexjs.org/guide/#configuration-options)` for better UI.

Comment (starbirdtech383) [reply packages/noco-docs/docs/100.data-sources/020.connect-to-data-source.md:32]: 1. "\" does add line break while rendering. For readability let me add it in source as well. 
2. ack. good catch.

File: packages/noco-docs/docs/100.data-sources/020.connect-to-data-source.md (hunk 1 above)
Comment (starbirdtech383) [reply packages/noco-docs/docs/100.data-sources/020.connect-to-data-source.md]: 1. ack, removing that extra \
2. 4a. was added considering a reference to image would. I am attaching this image now. 
image

File: packages/noco-docs/docs/100.data-sources/020.connect-to-data-source.md (hunk 2 above)
Comment (wingkwong) [reply packages/noco-docs/docs/100.data-sources/020.connect-to-data-source.md]: 1. maybe add one more. you can see the above screenshot. two paragraphs are packed to each other.

Comment (starbirdtech383) [reply packages/noco-docs/docs/100.data-sources/020.connect-to-data-source.md]: although not exactly what you suggested here. I slightly reworded to reduce verbosity. Below is how it looks now

source view : 
image

rendered view: 
image

Comment (wingkwong) [reply packages/noco-docs/docs/100.data-sources/020.connect-to-data-source.md]: Looks good now.

**Comment by starbirdtech383** [comment]
thanks for review!

**Pull request merged**